File: CarlaErik/UvA.SoftCon.Questionnaire/AST/Model/Expressions/Literal.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace UvA.SoftCon.Questionnaire.AST.Model.Expressions
{
    public abstract class Literal<T> : Node, IExpression
    {
        public T Value
        {
            get;
            private set;
        }

        protected Literal(T value, TextPosition position)
            : base(position)
        {
            Value = value;
        }

        public override string ToString()
        {
            return Value.ToString();
        }
    }
}

File: Stravopodis_Safaric/QL/src/uva/ql/ast/expressions/literals/Literal.java
package uva.ql.ast.expressions.literals;

import uva.ql.ast.CodeLines;
import uva.ql.ast.expressions.Expression;

public abstract class Literal extends Expression{

	public Literal(){
		super();
	}
	
	public Literal(CodeLines codeLines) {
		super(codeLines);
	}
	
	@Override
	public boolean isLiteral(){
		return true;
	}
	
	@Override
	public boolean isBinaryExpression(){
		return false;
	}
}

